sljit_src/sljitExecAllocator.c

```diff
@@ -182,10 +182,12 @@ static SLJIT_INLINE void* alloc_chunk(sljit_uw size)
 	if (retval == MAP_FAILED)
 		return NULL;
 
+#ifdef SLIJT_WX_OS_NEEDSCHEK
```

Comment:
HardenedBSD might be the only OS I am aware of that might have still an implementation of PAX that lies on their mmap() and need this, other than macOS (which also lies, but had already working code to flip permissions as needed since #90)

Reply:
Just to suggest a possible alternative, the `PROT_READ | PROT_WRITE | PROT_EXEC` bit could be a macro that is set to `PROT_READ | PROT_EXEC` for M1, because those are the actually permissions for the initial mmap. Then the mprotect check should work fine, right?

(This is just based on my understanding of how this works, I have no system to test.)

> there are no OS especific checks where this could be set yet, neither I am sure if the flag could have a better name so pushing this as a draft to discuss further before merging.

Alternatively could make this an `#ifndef SLJIT_WX_DONT_CHECK` that is defined in the M1 branch.